src/MPI.jl

```diff
@@ -97,6 +97,23 @@ include("mpiexec_wrapper.jl")
 include("deprecated.jl")
 
 function __init__()
+
+    # an empty string was used to indicate "default"
+    # https://github.com/JuliaParallel/MPI.jl/blob/v0.19.2/deps/build.jl#L142
+    mpi_env_binary = get(ENV, "JULIA_MPI_BINARY", "")
+    if mpi_env_binary != "" && mpi_env_binary != MPIPreferences.binary
+        @info """
+        The JULIA_MPI_BINARY environment variable is no longer used to configure the MPI binary.
+        Please use the MPIPreferences.jl package instead:
```

Comment:
What about playing back to the user what lead to the warning such that even unsuspecting users (e.g., on module-managed systems with automatic environment variables) know what to set their MPIPreferences to? For example, something like this:
```suggestion
        The JULIA_MPI_BINARY environment variable is no longer used to configure the MPI binary.
        This results in a mismatch between your environment variable setting ("$mpi_env_binary") and
        what MPI.jl actually uses ("$MPIPreferences.binary").
        
        Please use the MPIPreferences.jl package for configuration instead:
```

Reply:
that is actually what the keys below do: see my comment above.

Reply:
Great, obviously you were at least one step ahead of me 👍